tests/cpu_efm32_drivers/Makefile

```diff
@@ -0,0 +1,17 @@
+BOARD ?= sltb001a
+include ../Makefile.tests_common
+
+BOARD_WHITELIST := ikea-tradfri \
```

Comment:
Does this list contain all efm32 boards supported in RIOT ?
If yes, you could have used `FEATURES_REQUIRED += arch_efm32` instead of an hard coded list.

Reply:
Yes, they _currently_ contain all supported boards, but support depends on definitions in `board.h`.

Reply:
So the usual way to do this properly is to use the feature mechanism.

Anyway, this PR seems to address the same thing as #11693 but in a different way (using a regular driver module and not a periph). See https://github.com/RIOT-OS/RIOT/pull/11693#issuecomment-610272706 for some conclusion.

Too bad this got merged so fast without a proper analysis of the situation IMHO.

Reply:
I mean we could have

```C
#ifndef CORETEMP_ADC
#define CORETEMP_ADC        ADC_LINE(0)
#endif 
```

so we don't have to re-define this for every board.

> Too bad this got merged so fast without a proper analysis of the situation IMHO.

￼The API is not set in stone yet, we can change the function names easiely as a follow up.
nRF5x also has this feature `saul_nrf_temperature` so a common API would make sense.

Reply:
> Too bad this got merged so fast without a proper analysis of the situation IMHO.

I must admit that I have missed #11693. As I see, the difference is that #11693 provides a peripheral for reading the internal temperature, where this provides a driver on top of the ADC peripheral.

> so we don't have to re-define this for every board.

We do have to define this for every board, since you don't know which `ADC_LINE` is used for the internal temperature sensor (if defined at all).

Reply:
> We do have to define this for every board, since you don't know which ADC_LINE is used for the internal temperature sensor (if defined at all).

The I would suggest re-working this as a feature (as @aabadie  suggested) instead of a Module, that way boards that do have the configuration will be able to add `FEATURES_PROVIED += efm32_coretemp`, it will be cleaner to use outside of the test application, what do you think @basilfx?

Reply:
@fjmolinas makes sense. I opened https://github.com/RIOT-OS/RIOT/pull/15470 to fix it.